### PR TITLE
add edit functionality

### PR DESCRIPTION
**Pull request by melanie-adler**

### CODE REVIEW

File: src/Components/ColorInput/ColorInput.jsx [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-  const [InputValue, setInputValue] = useState(defaultValue);
+export default function ColorInput({ id, value, defaultValue, onChange }) {
+  const [InputValue, setInputValue] = useState(value);
 
-  function handleInputValue(event) {
+  useEffect(() => {
+    setInputValue(value);
+  }, [value]);
+
+  function handleChange(event) {
```
Comment (predrag-babic) [inline src/Components/ColorInput/ColorInput.jsx:10]: do you need useEffect here?

File: src/Components/EditButton/EditButton.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import { useState } from "react";
+import ColorInput from "../ColorInput/ColorInput";
+
+export default function EditButton({ color, onEditColor }) {
+  const [isEditing, setIsEditing] = useState(false);
+  const [editData, setEditData] = useState(color);
```
Comment (Leon910) [inline src/Components/EditButton/EditButton.jsx:6]: why do you use two useStates? isn´t isEditing enough to set the state of the Edit-button?

File: src/Components/ColorInput/ColorInput.jsx (hunk 1 above)
Comment (melanie-adler) [reply src/Components/ColorInput/ColorInput.jsx:10]: You're right. It's unnecessary. Remnant of something I tried before I settled on the code I have now. Good catch, thanks!

File: src/Components/EditButton/EditButton.jsx (hunk 2 above)
Comment (melanie-adler) [reply src/Components/EditButton/EditButton.jsx:6]: isEditing sets the state of the editing mode, editData is used to handle the data changes. I currently don't see another way to solve this. Any suggestions?

**Pull request merged**